extras/rya.streams/query-manager/src/main/java/org/apache/rya/streams/querymanager/QueryExecutor.java

```diff
@@ -40,18 +40,27 @@
      *
      * @param ryaInstanceName - The rya instance whose {@link Statement}s will be processed by the query. (not null)
      * @param query - The query to run. (not null)
+     * @throws QueryExecutorException When the query fails to start.
      */
-    public void startQuery(final String ryaInstanceName, final StreamsQuery query);
+    public void startQuery(final String ryaInstanceName, final StreamsQuery query) throws QueryExecutorException;
 
     /**
      * Stops a {@link StreamsQuery}.
      *
      * @param queryID - The ID of the query to stop. (not null)
+     * @throws QueryExecutorException When the query fails to stop.
      */
-    public void stopQuery(final UUID queryID);
+    public void stopQuery(final UUID queryID) throws QueryExecutorException;
 
     /**
      * @return - A set of {@link UUID}s representing the current active queries.
      */
     public Set<UUID> getRunningQueryIds();
```

Comment:
This should also throw the exception for when you can't discover what is running.